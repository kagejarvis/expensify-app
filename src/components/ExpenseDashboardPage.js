import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummaryOriginal';

const ExpenseDashboardPage = () => (
    <div>
        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;