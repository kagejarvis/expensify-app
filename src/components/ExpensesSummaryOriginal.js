import React from 'react';
import { connect } from 'react-redux';
import expenseCount from '../selectors/expenseCountOriginal';
import expensesTotal from '../selectors/expenses-totalOriginal';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export class ExpensesSummary extends React.Component {

    render() {
        return (
            <div>
                <h2>
                    Viewing {expenseCount(this.props.expenses)}
                    {
                        expenseCount(this.props.expenses) === 1 ? ` expense` : ` expenses`
                    } totalling {numeral((expensesTotal(this.props.expenses)) / 100).format('$0,0.00')}
                </h2>
            </div>

        );
    }
}


const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    }
};

export default connect(mapStateToProps)(ExpensesSummary);
