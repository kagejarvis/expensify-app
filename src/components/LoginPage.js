import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <img src="images/bank-icon.png" height="80"></img>
            <h1 className="box-layout__title">Expensify</h1>
            <p>tracking expenses made easy</p>
            <button className="button" onClick={startLogin}>Login with Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);