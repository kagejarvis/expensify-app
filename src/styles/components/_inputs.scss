.text-input {
    border: 2px solid #cacccd;
    height: 50px;
    font-size: $font-size-large;
    padding: $s-size;
}

.select {
    @extend .text-input;
}

.textarea {
    @extend .text-input;
    height: 10rem;
}

.DateRangePickerInput {
    border: 2px solid #cacccd !important;
    height: 50px !important;
}

.DateInput {
    padding: 0px !important;
    text-align: center;
}