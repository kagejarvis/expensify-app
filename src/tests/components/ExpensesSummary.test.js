import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should correctly render expenses summary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should correctly render expenses summary with mulitple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={1235} />);
    expect(wrapper).toMatchSnapshot();
});