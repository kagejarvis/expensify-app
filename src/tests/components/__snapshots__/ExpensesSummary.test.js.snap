// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render expenses summary with mulitple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $12.35
  </h1>
</div>
`;

exports[`Should correctly render expenses summary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h1>
</div>
`;
